instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/jdbc/OpenTelemetryJdbcDriverAutoConfiguration.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.instrumentation.jdbc;
+
+import io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver;
+import io.opentelemetry.instrumentation.spring.autoconfigure.OpenTelemetryInjector;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@ConditionalOnClass(OpenTelemetryDriver.class)
+@ConditionalOnProperty(
+    name = "spring.datasource.driver-class-name",
+    havingValue = "io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver")
+@Configuration(proxyBeanMethods = false)
```

Comment:
is this just an optimization, or needed for some other reason?

Reply:
It's an optimization (not for AOT and native modes) to avoid CGLIB processing.